bindings/pydeck-carto/setup.py

```diff
@@ -20,7 +20,8 @@
     python_requires=">=3.7",
     install_requires=[
         "pydeck>=0.8.0",
-        "carto-auth>=0.1.0",
+        "carto-auth>=0.2.0",
+        "typing-extensions>=4.0.0",
```

Comment:
`typing-extensions` is here to support type checking in Python versions < 3.10. Currently Pydeck supports Python 3.7+. As an alternative to the dependency, we could raise the minimum version, or do without the types.

Reply:
This is a good time to bump the version requirements. 3.8 is a no-brainer as 3.7 is end-of-life: https://devguide.python.org/versions/

Dumb question: Do we need to support type checking in 3.8 and 3.9? Or could we just not offer that feature in those versions and document the limitation?

Reply:
I looked more into which features of `typing-extensions` landed in which versions of Python, it turned out that while some typing features landed in 3.10, features we're using here landed in 3.11 and 3.12. Here are the versions for each, and what they do in pydeck:

- **typing.NotRequired**: Python 3.11+, allows us to mark specific options required vs. optional easily. Without this we would give up the ability for the type hints to catch missing required dependencies.
- **typing.Unpack**: Python 3.12+, Allows `**kwargs` to be typed, see [PEP 692](https://peps.python.org/pep-0692/). Without this we would either give up the type hints, or need to explicitly duplicate all options for each of the `foo_bar_source()` functions.
- **typing.assert_type**: Python 3.11+, convenient, but not that important to the PR.

It's [too early to require Python 3.12+](https://devguide.python.org/versions/). I'm not sure that it's possible to support type checking in newer versions of Python in a backward-compatible way, without either publishing separate packages (non-starter) or including `typing-extensions` to polyfill support, as I've done here. I'm new to type-checking in Python though, there may be options I've overlooked.

But I agree it makes sense to bump the minimum version from 3.7 → 3.8 at least, regardless of any of the above — I'll open a separate PR for that.